### DISCUSSION TITLE
Layouts using ChatGPT

### DISCUSSION THREAD

**Discussion by bwente** [HASwitchPlate/openHASP#499] [Show and tell]
I taught Chat-GPT how to do layouts.

<img width="895" alt="image" src="https://github.com/HASwitchPlate/openHASP/assets/371964/1f373423-7970-42bd-b36c-27176fe726fb">

**Comment by jazzmonger** [comment]
Ok, this is oddly intriguing! What else can it do?  Can you feed it existing pages and update a design?
Couple this with the designer from @mvturnho and what do you get?

**Reply by bwente** [reply]
**Yes, you need to train it then it can update you screens.**

<img width="851" alt="image" src="https://github.com/HASwitchPlate/openHASP/assets/371964/b6af26ef-91a5-4cc9-9424-153491a6788c">

**Comment by sammcj** [comment]
Have you managed to get chatGPT (or other ML) to generate some home assistant dynamic templates?

So I really want to get all my home assistant entities available to openHASP and was trying to figure out how to get chatGPT to generate a template but I had so many issues with home assistant's config files and templating - it did my head in.

**Reply by bwente** [reply]
**Yes you can. Just show it an example of a good yaml and it will get you pretty close.**

<img width="872" alt="image" src="https://github.com/HASwitchPlate/openHASP/assets/371964/5354cfc2-dd9e-41c4-bbd5-60ef4cf41872">







**The it ask it to make a new one.**

<img width="933" alt="image" src="https://github.com/HASwitchPlate/openHASP/assets/371964/413222ee-4ba6-49cc-95f2-f5f31d673e39">

**Reply by sammcj** [reply]
What about for generating a template/macro that generates the items based on entities / devices found within HA? That's something I've been trying to do for a while - jinja2 is really dreadful.
